docs:#1092 adjacent segment merging (#1200)
redpanda-data/documentation: docs/manage/tiered-storage.mdx

| `cloud_storage_segment_max_upload_interval_sec` | The number of seconds for idle timeout. If this property is empty, Redpanda uploads metadata to the cloud storage, but the segment is not uploaded until it reaches the `segment.bytes` size. By default, the property is empty. |
| `cloud_storage_trust_file` | The public certificate used to validate the TLS connection to cloud storage. If this is empty, Redpanda uses your operating system's CA cert pool. |

## Cloud storage housekeeping

> To improve performance and scalability, Redpanda performs cloud storage housekeeping when the system is idle. Redpanda constantly calculates cloud storage utilization using the moving average with a sliding window algorithm. The width of the sliding window is defined by the `cloud_storage_idle_timeout_ms` property. Default is 10 seconds. 

Reviewer: ```suggestion
To improve performance and scalability, Redpanda performs cloud storage housekeeping when the system is idle. Redpanda constantly calculates cloud storage utilization using the moving average with a sliding window algorithm. The width of the sliding window is defined by the `cloud_storage_idle_timeout_ms` property, which has a default of 10 seconds. 
```

Author: fixed

> If the utilization (requests per second) goes below the threshold, then cloud storage is considered idle, and cloud storage housekeeping begins. The threshold is defined by the `cloud_storage_idle_threshold_rps` property. Default is one request per second. Cloud storage is considered idle if, during the last 10 seconds, there were 10 or less cloud storage API requests.

Reviewer: ```suggestion
If the utilization (requests per second) drops below the threshold, then cloud storage is considered idle, and cloud storage housekeeping begins. The threshold is defined by the `cloud_storage_idle_threshold_rps` property, which as a default of one request per second. Cloud storage is considered idle if, during the last 10 seconds, there were 10 or less cloud storage API requests.
```

Author: fixed

If the cloud storage API becomes active after housekeeping begins, then housekeeping is paused until it becomes idle again. If cloud storage is not idle for `cloud_storage_housekeeping_interval_ms milliseconds`, then housekeeping is forced to run until it completes. This guarantees that all housekeeping jobs are run once for every `cloud_storage_housekeeping_interval_ms`. 

### Adjacent segment merging

> By default, as part of housekeeping, Redpanda runs adjacent segment merging on all segments in the cloud that are smaller than the threshold. Two properties control the behavior of `cloud_storage_enable_segment_merging`:

Reviewer: ```suggestion
By default, and as part of housekeeping, Redpanda runs adjacent segment merging on all segments in the cloud that are smaller than the threshold. Two properties control the behavior of `cloud_storage_enable_segment_merging`:
```

Author: fixed

> - `cloud_storage_segment_size_target`: This is the desired segment size in cloud storage. The default segment size is the local segment size for the topic, which is controlled by the `log_segment_size` configuration property and the `segment.bytes` topic configuration property. The `cloud_storage_segment_size_target` property can be set to a value larger than this default segment size, but because that triggers a lot of segment reuploads, it's not recommended. 

Reviewer: ```suggestion
- `cloud_storage_segment_size_target`: This is the desired segment size in cloud storage. The default segment size is the local segment size for the topic, which is controlled by the `log_segment_size` configuration property and the `segment.bytes` topic configuration property. You can set the `cloud_storage_segment_size_target` property to a value larger than this default segment size, but because that triggers a lot of segment reuploads, it's not recommended. 
```

Author: fixed
